packages/backend/src/Application.ts

```diff
@@ -59,15 +62,21 @@ export class Application {
       logger,
     )
     const eventRepository = new EventRepository(database, logger)
+    const txCountRepository = new TxCountRepository(database, logger)
 
     const http = new HttpClient()
 
-    const provider = new providers.AlchemyProvider(
+    const ethereumProvider = new providers.AlchemyProvider(
       'mainnet',
       config.alchemyApiKey,
     )
+    const optimismProvider = new providers.AlchemyProvider(
```

Comment:
I am a bit afraid that adding more and more providers will bloat the Application.ts code.
Maybe we should create one aggregate Provider which methods would take parameter {network: ethereum | optimism | etc.}

Reply:
or to not break to much of current logic just add MultiNetwork provider with network parametrization

Reply:
This is more a POC of this implementation and optimism is just an example here. So no, we will not create 10 more providers in the Application.ts code